speechbrain/utils/distributed.py

```diff
@@ -275,6 +274,13 @@ def ddp_init_group(run_opts):
             + " communication protocol doesn't exist."
         )
 
+    # Switch to the right context ; this is needed for DDP barriers
+    # otherwise, everything before an instance of `Brain(...)`
+    # is going to have `LOCAL_RANK` equal to 0.
+    local_rank = int(os.environ.get("LOCAL_RANK", "0"))
+    device = torch.device(f"cuda:{local_rank}")
+    torch.cuda.set_device(device)
```

Comment:
Looks like PyTorch docs do not recommend this func https://pytorch.org/docs/stable/generated/torch.cuda.set_device.html

Reply:
yes i saw it too... but at the same time I'm seeing everyone setting it this way (and this is what we are doing in core.py). I will look into that to see if there's better ways to do it.